### PR TITLE
Feature/review q2

### LINKED ISSUE #35: Add approval queue for new terms
Terms should be approved by a metadata expert at each institution.

### LINKED ISSUE #206: Terms/Vocabs/Predicates need to be set for review
The models need to know that they are under review so they dont display in the "real world". Add any necessary data to the record to know whether it is under review.

### LINKED ISSUE #208: Create review queue and hide from generic users
We need a page where the review process is done. It can be accessible from an admin dashboard or from a link in the nav bar. This needs to display records under review with a button to edit/reject/approve.

### PR DESCRIPTION
**Pull request by lsat12357**
fixes #35 fixes #208 fixes #206

### CODE REVIEW

File: app/controllers/concerns/git_interface.rb [hunk 1]
```diff
@@ -2,69 +2,101 @@ module GitInterface
...
+      #commit
       options = {}
-      options[:tree] = commit_tree
-      options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
-      options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
-      options[:message] ||= "Merge #{from_branch} into #{into_branch}"
-      options[:parents] = [repo.head.target, our_commit]
+      options[:tree] = index.write_tree(repo)
+      options[:author] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
+      options[:committer] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
```
Comment (straleyb) [inline app/controllers/concerns/git_interface.rb]: Im curious as to what the difference between the author and the committer is here. Just for clarification on my end.

**Comment by lsat12357** [comment]
I think they would always be the same person (in our case)

File: app/controllers/concerns/git_interface.rb [hunk 2]
```diff
@@ -2,69 +2,101 @@ module GitInterface
...
+
+      if merge_index.conflicts?
+        logger.error('merge conflict with branch: ' + branch_id)
+        raise 'merge conflict'
+      else
+        commit_tree = merge_index.write_tree(repo)
+        options = {}
+        options[:tree] = commit_tree
+        options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
+        options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
```
Comment (straleyb) [inline app/controllers/concerns/git_interface.rb]: These few lines keep reoccurring. Maybe we could split it into its own method to cut down on repetition? I see something like
```
def set_author_committer
     options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
     options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
end
```

**Comment by lsat12357** [comment]
totally agree; I have another branch that refactors that, and also refactors the walker stuff. But I figured I'd save it until after the new code (for review queue) is in?

**Comment by straleyb** [comment]
Yeah that sounds good :) Just as a heads up this branch looks like it needs rebased again.

File: app/controllers/concerns/git_interface.rb [hunk 3]
```diff
@@ -193,4 +225,117 @@ def format_response(results)
...
+    end
+    repo.checkout('master')
+    terms
+  end
+
+  #takes string and converts to rdf statement
+  def triple_string_to_statement(tstring)
+    RDF::Reader.for(:ntriples).new(tstring) do |reader|
+      reader.each_statement do |statement|
+        return statement
```
Comment (straleyb) [inline app/controllers/concerns/git_interface.rb]: Does this just return the first statement in the list, then leave the method?

**Comment by lsat12357** [comment]
yes.
<wrinkles brow>
hmm. it could be clearer that it only expects to return one statement.

**Comment by straleyb** [comment]
Yeah the thing is the #each_statement sort of thew me off. Could you do something like 
```
#statements.first 
```
Or something similar to that? Just a thought.

File: spec/controllers/terms_controller_spec.rb [hunk 4]
```diff
@@ -137,31 +140,36 @@ def get_new
           :comment => ["en"]}}
         }
       end
-      let(:save_success) { true }
+      #let(:save_success) { true }
```
Comment (straleyb) [inline spec/controllers/terms_controller_spec.rb]: If these arent going to be used we could probably remove them.

File: spec/controllers/terms_controller_spec.rb [hunk 5]
```diff
@@ -137,31 +140,36 @@ def get_new
...
+          post :create, params
+        end
         it "should require login" do
           expect(response.body).to have_content("Only admin can access")
         end
       end
-      it "should save term form" do
-        expect(term_form).to have_received(:save)
-      end
+      #it "should save term form" do
```
Comment (straleyb) [inline spec/controllers/terms_controller_spec.rb]: These too.

File: spec/controllers/terms_controller_spec.rb [hunk 6]
```diff
@@ -180,22 +188,29 @@ def get_new
...
-      context "when save fails" do
-        let(:save_success) { false }
-        it "should render new template" do
-          expect(response).to render_template("new")
-        end
-        it "should assign @term" do
-          expect(assigns(:term)).to eq term_form
-        end
-      end
+      #context "when save fails" do
```
Comment (straleyb) [inline spec/controllers/terms_controller_spec.rb]: And here.

**Comment by straleyb** [comment]
So last thing is I saw some commented out tests. I also poked greg to look at it too. Looks good though. Last thing that I can think of is just squashing your commits and id say its good to go!

File: app/controllers/predicates_controller.rb [hunk 7]
```diff
@@ -54,14 +57,72 @@ def deprecate
...
       @term = edit_predicate_form
       render "deprecate"
     end
   end
 
+
+  def commit
+    if Term.exists? params[:id]
+      predicate_form = predicate_form_repository.find(params[:id])
+      predicate_form.attributes = predicate_params
```
Comment (lsat12357) [inline app/controllers/predicates_controller.rb]: pretty sure this needs to be = vocabulary_params

File: app/controllers/vocabularies_controller.rb [hunk 8]
```diff
@@ -39,28 +42,81 @@ def update
...
+      vocabulary_form = vocabulary_form_repository.new(params[:id], Vocabulary)
+      vocabulary_form.attributes = vocabulary_params.except(:id, :issued)
+      action = "new"
+    end
+    vocabulary_form.set_languages(params[:vocabulary])
+    vocabulary_form.set_modified
+    vocabulary_form.reset_issued(params[:issued])
+
+    if vocabulary_form.is_valid?
+      triples = vocabulary_form.sort_stringify(vocabulary_form.full_graph)
```
Comment (lsat12357) [inline app/controllers/vocabularies_controller.rb]: this needs to be changed to single_graph

**Pull request closed without merging**